algolia/gatsby-plugin-algolia
# Migrate to algoliasearch@4.x

### What is the purpose of this pull request?

To update `algoliasearch` to the latest version, `4.9.1`, adapting the code per the [upgrade guide](https://www.algolia.com/doc/api-client/getting-started/upgrade-guides/javascript/) to take advantage of the latest improvements.

### What problem is this solving?

Not necessarily an issue, but letting this plugin rely on an outdated major certainly isn't ideal. Let me know what you think!

### How should this be manually tested?

- Install deps from both the root package and the `example` package.
- Create a brand new app on Algolia and open its indices page. It shouldn't have any indices (in case you followed this step)
- Set up your credentials on the `.env.production` on the `example` package.
- Run `yarn build` on the `example` package.
- Now refresh the indices page. You should see the pages from the `example` app there.

You can also take a look at one of my project's which is using an unpublished custom version of these changes, here: https://github.com/marcelovicentegc/marcelovicentegc.github.io/blob/main/gatsby-node.js#L9-L372.

To check it live: https://marcelo.page

### Screenshots or example usage

https://user-images.githubusercontent.com/33183880/118381932-70780d00-b5c6-11eb-999d-c1ca1e4b57e7.mov

### Related issues/pull requests

https://github.com/algolia/gatsby-plugin-algolia/issues/102
